aidial_adapter_vertexai/llm/gemini_chat_completion_adapter.py

```diff
@@ -0,0 +1,142 @@
+from typing import AsyncIterator, Dict, List, Tuple
+
+from aidial_sdk.chat_completion import Message
+from google.cloud.aiplatform_v1beta1.types import content as gapic_content_types
+from typing_extensions import override
+from vertexai.preview.generative_models import (
+    ChatSession,
+    GenerationConfig,
+    GenerativeModel,
+)
+
+from aidial_adapter_vertexai.llm.chat_completion_adapter import (
+    ChatCompletionAdapter,
+)
+from aidial_adapter_vertexai.llm.consumer import Consumer
+from aidial_adapter_vertexai.llm.gemini_prompt import GeminiPrompt
+from aidial_adapter_vertexai.llm.vertex_ai import (
+    get_gemini_model,
+    init_vertex_ai,
+)
+from aidial_adapter_vertexai.universal_api.request import ModelParameters
+from aidial_adapter_vertexai.universal_api.token_usage import TokenUsage
+from aidial_adapter_vertexai.utils.log_config import vertex_ai_logger as log
+from aidial_adapter_vertexai.utils.timer import Timer
+
+HarmCategory = gapic_content_types.HarmCategory
+HarmBlockThreshold = gapic_content_types.SafetySetting.HarmBlockThreshold
+
+BLOCK_NONE_SAFETY_SETTINGS: Dict[HarmCategory, HarmBlockThreshold] = {
+    HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_NONE,
+    HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_NONE,
+    HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_NONE,
+    HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_NONE,
+}
+
+
+def create_generation_config(params: ModelParameters) -> GenerationConfig:
+    return GenerationConfig(
+        max_output_tokens=params.max_tokens,
+        temperature=params.temperature,
+        stop_sequences=[params.stop]
+        if isinstance(params.stop, str)
+        else params.stop,
+        top_p=params.top_p,
+        candidate_count=1 if params.stream else params.n,
```

Comment:
As far as I can see, in the send_message_async multiple candidates aren't handled. Should the value always be 1?

Reply:
Fixed: set to 1, added a comment.